src/sys/statfs.rs

```diff
@@ -1,20 +1,665 @@
+pub const PROC_SUPER_MAGIC: FsType = FsType(libc::PROC_SUPER_MAGIC);
+#[cfg(all(target_os = "linux", not(target_env = "musl"), not(target_arch = "s390x")))]
+pub const QNX4_SUPER_MAGIC: FsType = FsType(libc::QNX4_SUPER_MAGIC);
+#[cfg(all(target_os = "linux", not(target_env = "musl"), not(target_arch = "s390x")))]
+pub const REISERFS_SUPER_MAGIC: FsType = FsType(libc::REISERFS_SUPER_MAGIC);
+#[cfg(all(target_os = "linux", not(target_env = "musl"), not(target_arch = "s390x")))]
+pub const SMB_SUPER_MAGIC: FsType = FsType(libc::SMB_SUPER_MAGIC);
+#[cfg(all(target_os = "linux", not(target_env = "musl"), not(target_arch = "s390x")))]
+pub const TMPFS_MAGIC: FsType = FsType(libc::TMPFS_MAGIC);
+#[cfg(all(target_os = "linux", not(target_env = "musl"), not(target_arch = "s390x")))]
+pub const USBDEVICE_SUPER_MAGIC: FsType = FsType(libc::USBDEVICE_SUPER_MAGIC);
+
+impl Statfs {
+    /// Magic code defining system type
+    #[cfg(not(any(
+        target_os = "openbsd",
+        target_os = "ios",
+        target_os = "macos"
+    )))]
+    pub fn filesystem_type(&self) -> FsType {
+        FsType(self.0.f_type)
+    }
+
+    /// Magic code defining system type
+    #[cfg(any(
+        target_os = "openbsd",
+        target_os = "ios",
+        target_os = "macos"
+    ))]
+    pub fn filesystem_type(&self) -> &str {
+        let c_str = unsafe { CStr::from_ptr(self.0.f_fstypename.as_ptr()) };
+        c_str.to_str().unwrap()
+    }
+
+    /// Optimal transfer block size
+    #[cfg(any(
+        target_os = "ios",
+        target_os = "macos",
+        target_os = "openbsd"
+    ))]
+    pub fn optimal_transfer_size(&self) -> libc::int32_t {
+        self.0.f_iosize
+    }
+
+    /// Optimal transfer block size
+    #[cfg(all(
+        target_os = "linux",
+        target_arch = "s390x"
+    ))]
+    pub fn optimal_transfer_size(&self) -> libc::uint32_t {
+        self.0.f_bsize
+    }
+
+    /// Optimal transfer block size
+    #[cfg(any(
+        target_os = "android",
+        all(
+            target_os = "linux",
+            target_env = "musl"
+        )
+    ))]
+    pub fn optimal_transfer_size(&self) -> libc::c_ulong {
+        self.0.f_bsize
+    }
+
+    /// Optimal transfer block size
+    #[cfg(all(
+        target_os = "linux",
+        not(target_env = "musl"),
+        not(target_arch = "s390x")
+    ))]
+    pub fn optimal_transfer_size(&self) -> libc::c_long {
+        self.0.f_bsize
+    }
+
+    /// Optimal transfer block size
+    #[cfg(target_os = "dragonfly")]
+    pub fn optimal_transfer_size(&self) -> libc::c_long {
+        self.0.f_iosize
+    }
+
+    /// Optimal transfer block size
+    #[cfg(target_os = "freebsd")]
+    pub fn optimal_transfer_size(&self) -> libc::uint64_t {
+        self.0.f_iosize
+    }
+
+    /// Size of a block
+    #[cfg(any(
+        target_os = "ios",
+        target_os = "macos",
+        target_os = "openbsd"
+    ))]
+    pub fn block_size(&self) -> libc::uint32_t {
```

Comment:
Both the FreeBSD and Linux man pages describe this field as the fragment size.  Should you change the name accordingly?